### ISSUE TITLE
Begin thinking about reserved names

### ISSUE THREAD

**Issue by DevBowser** [Scopeweb/MERN-CMS#11]
In moving ahead with #9, and once we start growing other functionality and potentially start clashing endpoints we should start thinking about implementing reserved name lists.

For example, if #9 adds `/first`,`/last` and `/count` endpoints to querying users, we could potentially have clash because I've chosen to allow users to be queried by the username like:

`/api/v1/users/<username>/`

So if the following query brings back the count of users:

`/api/v1/users/count/`

Then, a user created named `count` would cause some issues.

To sum up, we need a reserved names list that will apply to usernames and article slugs when being created by the end-user. To start:

```
['first', 'last', 'count']
```
